### ISSUE TITLE
Bug: brave engine

### ISSUE THREAD

**Issue by RaptaG** [searxng/searxng#2616]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
Repository: https://github.com/searxng/searxng
Branch: master
Version: 2023.8.9+c00c0c54

**How did you install SearXNG?**

**What happened?**

**How To Reproduce**

**Expected behavior**

**Screenshots & Logs**

**Additional context**

**Technical report**

Error
  * Error: searx.exceptions.SearxEngineTooManyRequestsException
  * Percentage: 50
  * Parameters: `('Too many request',)`
  * File name: `searx/search/processors/online.py:118`
  * Function: `_send_http_request`
  * Code: `response = req(params['url'], **request_args)`

**Comment by geckolinux** [comment]
I'm seeing this too in virtually all of the public instances I've tried, Brave seems to be consistently failing.

**Comment by RaptaG** [comment]
> I'm seeing this too in virtually all of the public instances I've tried, Brave seems to be consistently failing.

Some instances do work

**Comment by geckolinux** [comment]
> > I'm seeing this too in virtually all of the public instances I've tried, Brave seems to be consistently failing.
> 
> Some instances do work

I think it's just a function of the popularity of those instances, and there is a low percentage of Brave reliability in most instances' stats, so it looks like Brave simply has a very low rate-limiting threshold.

**Comment by return42** [comment]
> so it looks like Brave simply has a very low rate-limiting threshold.

yes, that seems to be the case, before PR #2599 brave was [deactivated by default](
https://github.com/searxng/searxng/pull/2599/files#diff-e95e3f454925bc2344d3cf538950b11a1255935b47a609c9589e6ac33d74802fL1836) .. now the engine is activated with which the threshold value is quickly exceeded.

I leave this issue open .. may be there comes the time I can reverse-engineer brave's WEB client more deeper in the hope to find a clue not to trigger the threshold.

**Comment by RaptaG** [comment]
So you don't see it fixed in the near future @return42 right?

**Comment by return42** [comment]
If possible, we will fix it ASAP .. but I don't know, if we can fix it and when I have time for (or someone other of the community) 

----

Updates from my analysis:

- brave sends a 403 with a HTML page in the response text / the HTML page contains a "Confirm you’re a human being" as shown in this issue --> https://community.brave.com/t/captcha-softlocks-browser-usability/501300

**Comment by return42** [comment]
# Updates from my analysis:

Brave sends a HTTP 403 (Too many request) with a HTML page in the response text / the HTML page contains a "Confirm you’re a human being" as shown in this ticket:

- [Captcha Softlocks Browser Usability](https://community.brave.com/t/captcha-softlocks-browser-usability/501300)

As described in the ticket; Neither the *"I'm not a robot"* button nor the answer to the captcha method under the link *"Switch to traditional captcha"* offer a sustainable solution --> with every further search query you get a 403 response again and the "I'm not a robot" game starts all over again.

The problem does not only affect SXNG instances, also normal users (who have sent the same search term too often) seem to be added to a block list and have to confirm that they are not a bot with every new request.

## Captcha & Cookie

If you answer the captcha, then the search sometimes works / but I could not find a regularity.

The `__Secure-sku#brave-search-captcha` cookie seems to me to be the answer to the captcha, but that alone does not seem to be enough.  I have the impression that additionally a `session#brave-search-captcha` cookie changing with each request is forwarded from one response to the next request.

![image](https://github.com/searxng/searxng/assets/554536/42c02524-14e1-4491-af2a-7e33b1355d1f)

------

## Response from Brave in the `<noscript>` tag ..


<h5 class="text-center title svelte-1t2ctke">This page needs
JavaScript to function.</h5> Your request has been flagged
as being suspicious and Brave Search decided to schedule a
captcha for you. Unfortunately, <strong>your browser does not
seem to have JavaScript enabled.</strong> <br> <br> Here are a
few things you can do: <ul><li>If you are using the <strong>Brave
browser</strong> with scripts blocked, you can enable scripts
again for Brave Search then reload this page.  </li> <li>If you
are using the <strong>Tor browser</strong> with its <a
href="https://tb-manual.torproject.org/security-settings/#safest"
rel="noopener"><strong>safest</strong></a> security level, you
can selectively enable JavaScript on Brave Search via the
noscript extension toggle then reload this page.  </li> <li>If
the above does not match your situation, make sure that your
browser supports JavaScript and that no extension is currently
disabling JavaScript on Brave Search.  </li></ul></div>

**Comment by return42** [comment]
The brave engine does no longer get results / nut sure if API has changed / needs a new analysis .. *phew .. brave is really exhausting*

**Comment by PatyaRishim** [comment]
I wish I could help more but at least, I think, this is better than nothing. Apparently the last update fixed the brave engine but only in this instance so far (or as far as I have researched)

https://search.cronobox.one/

Maybe you can contact said instance maintainer and ask what is the config. there.

**Comment by geckolinux** [comment]
@PatyaRishim It was probably just a matter of timing, on `search.cronobox.one` Brave is currently throwing "Suspended: too many requests".

**Comment by return42** [comment]
Issue is from Aug 9, 2023 .. I can't reproduce a`searx.exceptions.SearxEngineTooManyRequestsException` exception these days .. I close this issue .. if you see this error again, please create a new report .. thanks 👍 

FYI: today we fixed:

- https://github.com/searxng/searxng/pull/4355
- https://github.com/searxng/searxng/pull/4356

**Comment by yoshyv** [comment]
@return42 

I just got the same error after searching on one of my private instances for less than 10 times. Searching on brave is not suspended if I open brave on my local browser from the same IP address.

```
<h2>Errors and exceptions</h2>
                    
Exception | searx.exceptions.SearxEngineTooManyRequestsException | Percentage | 100
-- | -- | -- | --
Too many request
searx/search/processors/online.py:116
_send_http_request
response = req(params['url'], **request_args)
```

However, on another private instance, I had no such issue after searching for a dozen of times.

Both are installed using docker compose and they're of the same version 2025.7.3+34728ae 
They are located in different countries (US v.s. Canada) but the former (problematic one) is dual stack while the latter is IPv4 only.
